### PR TITLE
Add backup extension resources

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:
This PR add the backup extension resources to Gardener as per (backup infrastructure proposal)[https://github.com/gardener/gardener/blob/master/docs/proposals/02-backupinfra.md]
 Ref: https://github.com/gardener/etcd-backup-restore/issues/141
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
This PR simply adds support for new extension resource. The migration logic from old backupInfra resource and actual extension creation logic will follow in separate PR.

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/apis/extensions/v1alpha1/types_backupentry.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+// BackupEntrySpec is the spec for an BackupEntry resource.
+type BackupEntrySpec struct {
+	// DefaultSpec is a structure containing common fields used by all extension resources.
+	DefaultSpec `json:",inline"`
+	// Region is the region of this Entry.
+	Region string `json:"region"`
+	// Bucket is the backup bucket for this Backup Entry.
+	Bucket string `json:"bucket"`
```
Comment (rfranzke) [inline pkg/apis/extensions/v1alpha1/types_backupentry.go]:
```suggestion
	BucketName string `json:"bucketName"`
```

That's the name of a `BackupBucket` resource, right?

File: pkg/apis/extensions/v1alpha1/types_backupentry.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	// SecretRef is a reference to a secret that contains the credentials to access object store.
+	SecretRef corev1.SecretReference `json:"secretRef"`
```
Comment (rfranzke) [inline pkg/apis/extensions/v1alpha1/types_backupentry.go:61]: Why is this needed? Can't we use the secret from the `BackupBucket`?

Comment (swapnilgm) [reply pkg/apis/extensions/v1alpha1/types_backupentry.go:61]: Actually yes, we can use the same. Initially I thought its better to keep Backup Entry unaware of BackupBucket. And let it be self contained with Region and access secret. I will remove that - region and secretRef.

Comment (rfranzke) [reply pkg/apis/extensions/v1alpha1/types_backupentry.go:61]: OK, but you anyway already have the bucket inside, so it's not unaware anymore :)

Comment (swapnilgm) [reply pkg/apis/extensions/v1alpha1/types_backupentry.go:61]: Ahh. We need secretRef and Region on extension/backupEntry but not on gardener/backupEntry, cause there will only on extension/backupBucket per region  in cloud profile, irrespective number of seeds for that region. So, in case the shoot is not scheduled on seed on which extension/backupBucket is scheduled, extension/backupEntry won't be able to find backupBucket and hence secret.
Said that we don't need region and secretRef on gardener/backupEntry since there it can get the details from gardener/backupBucket.

Comment (rfranzke) [reply pkg/apis/extensions/v1alpha1/types_backupentry.go:61]: OK, thank you, got it. Makes sense.

**Review by rfranzke — APPROVED** [review]
/lgtm
